packages/core-api/src/routing/BackstageRoutes.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode, Children, isValidElement, Fragment } from 'react';
+import { useRoutes } from 'react-router-dom';
+
+type RouteObject = {
+  path: string;
+  element: JSX.Element;
+  children?: RouteObject[];
+};
+
+// Similar to the same function from react-router, this collects routes from the
+// children, but only the first level of routes
+function createRoutesFromChildren(children: ReactNode): RouteObject[] {
+  return Children.toArray(children)
+    .flatMap(child => {
+      if (!isValidElement(child)) {
+        return [];
+      }
+
+      if (child.type === Fragment) {
+        return createRoutesFromChildren(child.props.children);
+      }
+
+      const { children } = child.props;
+
+      let path = child.props.path as string | undefined;
+
+      // TODO(Rugvip): Work around plugins registering empty paths, remove once deprecated routes are gone
+      if (path === '') {
+        return [];
+      }
+      path = path?.replace(/\/\*$/, '') ?? '/';
+
+      return [
+        {
+          path,
+          element: child,
+          children: children && [
+            {
+              path: '/*',
+              element: children,
+            },
+          ],
+        },
+      ];
+    })
+    .sort((a, b) => b.path.localeCompare(a.path))
+    .map(obj => {
+      obj.path = obj.path === '/' ? '/' : `${obj.path}/*`;
```

Comment:
Hm I'm wondering if we need to explicitly delete (or forbid) trailing slashes in path props so we don't get double slashes here?

Reply:
`/*` suffixes are trimmed above, and react-router trims duplicate slashes